nohost/src/HTMLServer.js

```diff
@@ -0,0 +1,104 @@
+
+    var Path = Filer.Path;
+
+    function HTMLServer(config) {
+        config = config || {};
+        BaseServer.call(this, config);
+    }
+
+    HTMLServer.prototype = Object.create(BaseServer.prototype);
+    HTMLServer.prototype.constructor = HTMLServer;
+
+    //Returns a pre-generated blob url based on path
+    HTMLServer.prototype.pathToUrl = function(path) {
+        return BlobUtils.getUrl(path);
+    };
+    //Returns a path based on blob url
+    HTMLServer.prototype.urlToPath = function(url) {
+        return BlobUtils.getFilename(url);
+    };
+
+    HTMLServer.prototype.start = function() {
+        this.fs = Filer.fs();
+    };
+
+    HTMLServer.prototype.stop = function() {
+        this.fs = null;
+    };
+
+    /**
+     * Determines if this server can serve local file. LiveDevServerManager
+     * calls this method when determining if a server can serve a file.
+     * @param {string} localPath A local path to file being served.
+     * @return {boolean} true When the file can be served, otherwise false.
+     */
+    HTMLServer.prototype.canServe = function (localPath) {
+        // If we can't transform the local path to a project relative path,
+        // the path cannot be served
+        if (localPath === this._pathResolver(localPath)) {
+            return false;
+        }
+
+        // Url ending in "/" implies default file, which is usually index.html.
+        // Return true to indicate that we can serve it.
+        if (localPath.match(/\/$/)) {
+            return true;
+        }
+
+        // FUTURE: do a MIME Type lookup on file extension
+        return LiveDevelopmentUtils.isStaticHtmlFileExt(localPath);
+    };
+
+    /**
+     * When a livedocument is added to the server cache, make sure live
+     * instrumentation is enabled
+     */
+    HTMLServer.prototype.add = function (liveDocument) {
+        if (liveDocument.setInstrumentationEnabled) {
+            // enable instrumentation
+            liveDocument.setInstrumentationEnabled(true);
+        }
+        BaseServer.prototype.add.call(this, liveDocument);
+    };
+
+    /**
+     * If a livedoc exists, serves the instrumented version of the file as as a blob URL.
+     * Otherwise, it serves only the file's contents as a blob URL.
+     */
+    HTMLServer.prototype.serveLiveDoc = function(url, callback) {
+        var path = BlobUtils.getFilename(url);
+        var liveDocument = this._liveDocuments[path];
+
+        function toURL(err, html) {
+            if (err) {
+                callback(err);
+                return;
+            }
+
+            // Convert rewritten HTML to a Blob URL Object
+            var url = Content.toURL(html, "text/html");
+
+            callback(null, url);
+        }
+
+        // If we have a LiveDoc for this path, send instrumented response. Otherwise fallback to static file from fs
+        if (liveDocument && liveDocument.getResponseData) {
```

Comment:
Is there ever a case where this `if` will be false?  Do we actually need to do this check?  What if it fails, what should happen?  Right now we're not calling the callback in that case.  My guess is that you can just remove the `if` check here and always do the rewriting.  Also, get rid of `toURL` and make it an inline anonymous function like this:

```js
Rewriter.rewrite(path, liveDocument.getResponseData().body, function(err, html) {
  ...
});
```